docs/refs/crd-upgrade-safety.md

```diff
@@ -0,0 +1,240 @@
+not cause the CRD Upgrade Safety preflight check to halt the upgrade:
+
+- Adding new enum values to the list of allowed enum values in a field
+- An existing required field is changed to optional in an existing version
+- The minimum value of an existing field is decreased in an existing version
+- The maximum value of an existing field is increased in an existing version
+- A new version of the CRD is added with no modifications to existing versions
+
+
+## Disabling CRD Upgrade Safety
+
+The CRD Upgrade Safety preflight check can be entirely disabled by adding the
+`preflight.crdUpgradeSafety.disabled` field with a value of "true" to the ClusterExtension of the CRD.
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+    name: clusterextension-sample
+spec:
+    installNamespace: default
+    packageName: argocd-operator
+    version: 0.6.0
+    preflight:
+        crdUpgradeSafety:
+            disabled: true
+```
+
+You cannot disable individual field validators. If you disable the CRD Upgrade Safety preflight check, all field validators are disabled.
+
+!!! warning
+    Disabling the CRD Upgrade Safety preflight check could break backwards compatibility with previous
+    versions of the CRD and cause other unintended consequences on the cluster.
+
+
+## Examples of Unsafe CRD Changes
+
+Take the following CRD as our starting version:
+
+```yaml
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.13.0
+  name: example.test.example.com
+spec:
+  group: test.example.com
+  names:
+    kind: Sample
+    listKind: SampleList
+    plural: samples
+    singular: sample
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        properties:
+          apiVersion:
+            type: string
+          kind:
+            type: string
+          metadata:
+            type: object
+          spec:
+            type: object
+          status:
+            type: object
+          pollInterval:
+            type: string
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+```
+
+The following examples will demonstrate specific changes to sections of the example CRD
+that would be caught by the CRD Upgrade Safety preflight check.
+
+### Changing Scope
+
+In this example, `scope` has been changed from `Namespaced` to `Cluster`.
+
+??? example
+    ```yaml
+    spec:
+      group: test.example.com
+      names:
+        kind: Sample
+        listKind: SampleList
+        plural: samples
+        singular: sample
+      scope: Cluster
+      versions:
+      - name: v1alpha1
+    ```
+
+??? failure "Error output"
+    ```
+    validating upgrade for CRD "test.example.com" failed: CustomResourceDefinition test.example.com failed upgrade safety validation. "NoScopeChange" validation failed: scope changed from "Namespaced" to "Cluster"
```

Comment:
I agree that the long line is a little hard to read, but I have always been told to keep example outputs as they are printed in the terminal.